tests/lib/rules/no-direct-mutation-state.js

```diff
@@ -70,6 +70,14 @@ ruleTester.run('no-direct-mutation-state', rule, {
       '}'
     ].join('\n')
   }, {
+    code: [
+      'class Hello extends React.Component {',
+      '  constructor() {',
+      '    this.state.foo = 1;',
```

Comment:
I don't see why it should be legal; it's mutating `this.state` after it's assigned.